validator/src/consensus/protocol/base.ts

```diff
@@ -58,9 +56,12 @@ export abstract class BaseProtocol implements ShieldnetProtocol {
 		const executionDelay = action.retryCount > 0 ? ERROR_RETRY_DELAY : 0;
 		setTimeout(() => {
 			this.performAction(action)
+				.then(() => {
+					// If action was successfully executed, remove it from queue
+					this.#actionQueue.pop();
+				})
 				.catch(() => {
 					action.retryCount++;
```

Comment:
Doesn't this only update in memory and not write to the DB?

Reply:
hmm yeah, will need to fix this